python/ray/job_config.py

```diff
@@ -1,81 +1,138 @@
 import uuid
-from typing import Any, Dict, List, Optional, Union
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
 
 import ray._private.gcs_utils as gcs_utils
 from ray.util.annotations import PublicAPI
 
+if TYPE_CHECKING:
+    from ray.runtime_env import RuntimeEnv
+
 
 @PublicAPI
 class JobConfig:
     """A class used to store the configurations of a job.
 
-    Attributes:
+    Args:
+
         jvm_options: The jvm options for java workers of the job.
         code_search_path: A list of directories or jar files that
             specify the search path for user code. This will be used as
             `CLASSPATH` in Java and `PYTHONPATH` in Python.
-        runtime_env: A runtime environment dictionary (see
-            ``runtime_env.py`` for detailed documentation).
-        client_job: A boolean represent the source of the job.
-        default_actor_lifetime: The default value of actor lifetime.
-        py_driver_sys_path: A list of directories that
-            specify the search path for python workers.
+            See :ref:`Ray cross-language programming <cross_language>` for more details.
+        runtime_env: A :ref:`runtime environment <runtime-environment>` dictionary.
+        metadata: An opaque metadata dictionary.
+        ray_namespace: A :ref:`namespace <namespaces-guide>`
+            is a logical grouping of jobs and named actors.
+        default_actor_lifetime: The default value of actor lifetime,
+            can be "detached" or "non_detached".
```

Comment:
I probably missed it, but where are the terms "detached" and "non-detached" actor defined?

Reply:
See `set_default_actor_lifetime()`.